app/Clans.php

```diff
@@ -22,9 +23,16 @@ public function getColorTagByTag($tag = null)
         return isset($tag) ? $this->getTable()->where('tag', '=', $tag)->get('color_tag')->first() : null;
     }
 
-    public function getHTMLColorTagByTag($tag = null)
+    public function getHTMLColorTagByTag($tag = null): ?string
     {
-        return isset($tag) ? $this->addColors($this->getTable()->where('tag', '=', $tag)->get('color_tag')->first()->color_tag) : null;
+        if (!isset($tag)) {
```

Comment:
_Determine if a variable is considered set, this means if a variable is declared and is different than null._ 
So, you check here: if the variable doesn't declared and is null.
How about:
```php
if (isset($tag)) {
  $row = $this->getTable()->where('tag', '=', $tag)->get('color_tag')->first();
  if (isset($row)) {
    return $this->addColors($row->color_tag);
  }
 }
 return null;
 ```
 
 Another way is:
 ```php
 if (isset($tag)) {
  $row = $this->getTable()->where('tag', '=', $tag)->get('color_tag');
  if ($row->isNotEmpty()) {
    return $this->addColors($row->first()->color_tag);
  }
 }
  return null;
 ```

Reply:
That is early return. If the tag passed is null, it returns.

Reply:
Early return avoids nested ifs.